OSDOCS-9695: Create the initial page for AWS AMIs in ROSA (#72483)
openshift/openshift-docs: modules/installation-aws-hcp-sd-ami-stream-metadata.adoc

//TODO: Add the module include to the following assemblies
//TODO: Create related modules for OpenStack (QCOW2) and Bare Metal (ISO)

// Module included in the following assemblies:
//
// * rosa_architecture/rosa-aws-amis.adoc

:_mod-docs-content-type: PROCEDURE
[id="installation-aws-hcp-sd-ami-stream-metadata_{context}"]

> = Accessing ROSA AMIs with stream metadata

Reviewer: ```suggestion
= Using stream metadata to access ROSA AMIs
```

Reviewer: Should you spell out AMI at first mention? And ROSA?

> In {product-title}, _stream metadata_ provides standardized metadata about RHCOS in the JSON format and injects the metadata into the cluster. Stream metadata is a stable format that supports multiple architectures and is intended to be self-documenting for maintaining automation.

Reviewer: Do RHCOS and/or JSON need to be spelled out at first mention?

> In {product-title}, _stream metadata_ provides standardized metadata about RHCOS in the JSON format and injects the metadata into the cluster. Stream metadata is a stable format that supports multiple architectures and is intended to be self-documenting for maintaining automation.

Reviewer: ```suggestion
In {product-title}, _stream metadata_ injects standardized metadata about RHCOS into the cluster in JSON format. Stream metadata is a stable format that supports multiple architectures and is self-documenting to maintain automation.
```

Reviewer: The above is a best guess. Take or leave it.

> You can use the `coreos print-stream-json` sub-command of `openshift-install` to access information about the boot images in the stream metadata format. This command provides a method for printing stream metadata in a scriptable, machine-readable format.

Reviewer: ```suggestion
Use the `coreos print-stream-json` subcommand of the `openshift-install` command to access information about the boot images in stream metadata format. This command prints stream metadata in a scriptable, machine-readable format.
```

> For user-provisioned installations, the `openshift-install` binary contains references to the version of RHCOS boot images that are tested for use with {product-title}, such as the AWS AMI.

Reviewer: ```suggestion
For user-provisioned installations, the `openshift-install` binary references the version of RHCOS boot images that are tested for use with {product-title}, such as the AWS AMI.
```

.Procedure

To parse the stream metadata, use one of the following methods:

> * From a Go program, use the official `stream-metadata-go` library at https://github.com/coreos/stream-metadata-go. You can also view example code in the library.

Reviewer: I think it would be better to make the website a link rather than write it out.

* From another programming language, such as Python or Ruby, use the JSON library of your preferred programming language.

> * From a command-line utility that handles JSON data, such as `jq`:

Reviewer: ```suggestion
* From a command line interface (CLI) that handles JSON data, such as `jq`:
```

** Print the current `x86_64` AMI for an AWS region, such as `us-west-1`:
+

> .For x86_64

Reviewer: I'm not sure what you are trying to say here. Are you trying to say: "To obtain/print the `x86_64` of a specific region, run the following command:"
